test/Microsoft.Azure.Devices.Edge.Test.Common/ProcessEx.cs

```diff
@@ -0,0 +1,127 @@
+                }
+            };
+
+            var standardErrorResults = new TaskCompletionSource<string[]>();
+            process.ErrorDataReceived += (sender, args) =>
+            {
+                if (args.Data != null)
+                {
+                    standardError.Add(args.Data);
+                    onStandardError(args.Data);
+                }
+                else
+                {
+                    standardErrorResults.SetResult(standardError.ToArray());
+                }
+            };
+
+            var processStartTime = new TaskCompletionSource<DateTime>();
+
+            process.Exited += async (sender, args) =>
+            {
+                // Since the Exited event can happen asynchronously to the output and error events,
+                // we await the task results for stdout/stderr to ensure they both closed.  We must await
+                // the stdout/stderr tasks instead of just accessing the Result property due to behavior on MacOS.
+                // For more details, see the PR at https://github.com/jamesmanning/RunProcessAsTask/pull/16/
+                tcs.TrySetResult(
+                    new ProcessResults(
+                        process,
+                        await processStartTime.Task.ConfigureAwait(false),
+                        await standardOutputResults.Task.ConfigureAwait(false),
+                        await standardErrorResults.Task.ConfigureAwait(false)));
+            };
+
+            using (cancellationToken.Register(
+                () =>
+                {
+                    tcs.TrySetCanceled();
+                    try
+                    {
+                        if (!process.HasExited)
+                            process.Kill();
+                    }
+                    catch (InvalidOperationException)
+                    {
+                    }
+                }))
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var startTime = DateTime.Now;
+                if (process.Start() == false)
+                {
+                    tcs.TrySetException(new InvalidOperationException("Failed to start process"));
+                }
+                else
+                {
+                    try
+                    {
+                        startTime = process.StartTime;
+                    }
+                    catch (Exception)
+                    {
+                        // best effort to try and get a more accurate start time, but if we fail to access StartTime
+                        // (for instance, process has already existed), we still have a valid value to use.
+                    }
+
+                    processStartTime.SetResult(startTime);
+
+                    process.BeginOutputReadLine();
+                    process.BeginErrorReadLine();
+                }
+
+                return await tcs.Task.ConfigureAwait(false);
+            }
+        }
+    }
+}
```

Comment:
Since this is the forked project you mentioned, it might help readability if we make a folder with its own namespace rather than dumping it in `Edge.Test.Common`.

Reply:
Yeah, we could, although I don't think it makes much difference either way.